src/components/networked.js

```diff
@@ -287,8 +275,14 @@ AFRAME.registerComponent('networked', {
       var syncedComponentData = componentSchema.property ? componentData[componentSchema.property] : componentData;
 
       // Use networkUpdatePredicate to check if the component needs to be updated.
-      // Call networkUpdatePredicate first so that it can update any cached values in the event of a fullSync.
-      if (this.networkUpdatePredicates[i](syncedComponentData) || fullSync) {
+      const previousData = this.previousSyncedComponentData[i];
+      const hasChanged = previousData === null || this.networkUpdatePredicates[i](previousData, syncedComponentData);
+
+      if (hasChanged) {
+        this.previousSyncedComponentData[i] = AFRAME.utils.clone(syncedComponentData);
+      }
```

Comment:
I think the point of this whole exercise was to track prior data on a type-by-type basis, so that we can do something more efficient than cloning every time (e.g. for vectors we want an implementation where each vector property has a single prior `Vector3` that it reuses every network frame.)